src/Task/Base/ParallelExec.php

```diff
@@ -165,14 +176,32 @@ public function run()
                 if (!$process->isRunning()) {
                     $this->advanceProgressIndicator();
                     if ($this->isPrinted) {
-                        $this->printTaskInfo("Output for {command}:\n\n{output}", ['command' => $process->getCommandLine(), 'output' => $process->getOutput(), '_style' => ['command' => 'fg=white;bg=magenta']]);
-                        $errorOutput = $process->getErrorOutput();
-                        if ($errorOutput) {
-                            $this->printTaskError(rtrim($errorOutput));
+                        if ($this->isWaitToFinishPrinted) {
```

Comment:
This method is already very cumbersome, and the extra length and conditionals is making it more so. Perhaps some refactoring could help here.

One thing you might consider is, instead of using a boolean `$this->isWaitToFinishPrinted` & c., maybe initialize a function pointer e.g. `$this->handleOutput = [$this, 'incrementalOutput'];`. Then the actual exec loop could just call the method pointer without needing a conditional.